File: WPF Databinding in Visual Studio 2010 RTM/[VB]-WPF Databinding in Visual Studio 2010 RTM/VB/WpfEfSampleApp/SimpleConverter.vb
Public Class SimpleConverter
    Implements IValueConverter

    Public Function Convert(ByVal value As Object, _
                            ByVal targetType As System.Type, _
                            ByVal parameter As Object, _
                            ByVal culture As System.Globalization.CultureInfo) As Object _
                            Implements System.Windows.Data.IValueConverter.Convert

        If parameter IsNot Nothing Then
            Return String.Format(culture, parameter.ToString(), value)
        End If

        Return value

    End Function

    Public Function ConvertBack(ByVal value As Object, _
                                ByVal targetType As System.Type, _
                                ByVal parameter As Object, _
                                ByVal culture As System.Globalization.CultureInfo) As Object _
                                Implements System.Windows.Data.IValueConverter.ConvertBack

        If targetType Is GetType(Date) OrElse targetType Is GetType(Nullable(Of Date)) Then
            If IsDate(value) Then
                Return CDate(value)
            ElseIf value.ToString() = "" Then
                Return Nothing
            Else
                Return Now() 'invalid type was entered so just give a default.
            End If

        ElseIf targetType Is GetType(Integer) OrElse targetType Is GetType(Nullable(Of Integer)) Then
            If IsNumeric(value) Then
                Return CInt(value)
            ElseIf value.ToString() = "" Then
                Return Nothing
            Else
                Return 0
            End If

        ElseIf targetType Is GetType(Decimal) OrElse targetType Is GetType(Nullable(Of Decimal)) Then
            If IsNumeric(value) Then
                Return CDec(value)
            ElseIf value.ToString() = "" Then
                Return Nothing
            Else
                Return 0
            End If
        End If

        Return value
    End Function
End Class

File: WPF Databinding in Visual Studio 2010 RTM/[C#]-WPF Databinding in Visual Studio 2010 RTM/C#/WpfEfSampleApp/SimpleConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Data;

namespace WpfEfSampleApp
{
    class SimpleConverter : IValueConverter
    {
        public object Convert(  object value,
                                System.Type targetType,
                                object parameter,
                                System.Globalization.CultureInfo culture)
        {
            if (parameter != null)
                return String.Format(culture, parameter.ToString(), value);

            return value;
        }

        public object ConvertBack(object value, System.Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            if ((targetType == typeof(DateTime)) || (targetType == typeof(Nullable<DateTime>)))
            {
                return System.Convert.ToDateTime(value);
            }

            else if ((targetType == typeof(int)) || (targetType == typeof(Nullable<int>)))
            {
                return System.Convert.ToInt32(value);
            }

            else if ((targetType == typeof(Decimal)) || (targetType == typeof(Nullable<Decimal>)))
            {
                return System.Convert.ToDecimal(value);
            }

            return value;
        }
    }
}
